Quetoo.vs15/libs/sdl_sound/SDL2/SDL_sound.h

```diff
@@ -0,0 +1,737 @@
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+
+/**
+ * \mainpage SDL_sound
+ *
+ * The latest version of SDL_sound can be found at:
+ *     http://icculus.org/SDL_sound/
+ *
+ * The basic gist of SDL_sound is that you use an SDL_RWops to get sound data
+ *  into this library, and SDL_sound will take that data, in one of several
+ *  popular formats, and decode it into raw waveform data in the format of
+ *  your choice. This gives you a nice abstraction for getting sound into your
+ *  game or application; just feed it to SDL_sound, and it will handle
+ *  decoding and converting, so you can just pass it to your SDL audio
+ *  callback (or whatever). Since it gets data from an SDL_RWops, you can get
+ *  the initial sound data from any number of sources: file, memory buffer,
+ *  network connection, etc.
+ *
+ * As the name implies, this library depends on SDL: Simple Directmedia Layer,
+ *  which is a powerful, free, and cross-platform multimedia library. It can
+ *  be found at http://www.libsdl.org/
+ *
+ * Support is in place or planned for the following sound formats:
+ *   - .WAV  (Microsoft WAVfile RIFF data, internal.)
+ *   - .VOC  (Creative Labs' Voice format, internal.)
+ *   - .MP3  (MPEG-1 Layer 3 support, via libmpg123.)
+ *   - .MID  (MIDI music converted to Waveform data, internal.)
+ *   - .MOD  (MOD files, via MikMod and ModPlug.)
+ *   - .OGG  (Ogg files, via Ogg Vorbis libraries.)
+ *   - .SPX  (Speex files, via libspeex.)
+ *   - .SHN  (Shorten files, internal.)
+ *   - .RAW  (Raw sound data in any format, internal.)
+ *   - .AU   (Sun's Audio format, internal.)
+ *   - .AIFF (Audio Interchange format, internal.)
+ *   - .FLAC (Lossless audio compression, via libFLAC.)
+ *
+ *   (...and more to come...)
+ *
+ * Please see the file LICENSE.txt in the source's root directory.
+ *
+ * \author Ryan C. Gordon (icculus@icculus.org)
+ * \author many others, please see CREDITS in the source's root directory.
+ */
+
+#ifndef _INCLUDE_SDL_SOUND_H_
+#define _INCLUDE_SDL_SOUND_H_
+
+#include "SDL.h"
+#include "SDL_endian.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifndef DOXYGEN_SHOULD_IGNORE_THIS
+
+#ifndef SDLCALL  /* may not be defined with older SDL releases. */
+#define SDLCALL
+#endif
+
+#ifdef SDL_SOUND_DLL_EXPORTS
+#  define SNDDECLSPEC __declspec(dllexport)
+#elif (__GNUC__ >= 3)
+#  define SNDDECLSPEC __attribute__((visibility("default")))
+#else
+#  define SNDDECLSPEC
+#endif
+
+#define SOUND_VER_MAJOR 1
+#define SOUND_VER_MINOR 0
+#define SOUND_VER_PATCH 1
```

Comment:
This is an older version of SDL_sound. I think you want 1.0.3. In version 1.0.3, `Sample_NewSampleFromMem` is not public. The "right way" to do this would be to create an `SDL_RWops` with `SDL_RWFromMem`, and then pass that to `Sample_NewSample`.